packages/opentelemetry-core/src/correlation-context/propagation/HttpCorrelationContext.ts

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Context,
+  CorrelationContext,
+  GetterFunction,
+  HttpTextPropagator,
+  SetterFunction,
+} from '@opentelemetry/api';
+
+import {
+  getCorrelationContext,
+  setCorrelationContext,
+} from '../correlation-context';
+
+export const CORRELATION_CONTEXT_HEADER = 'otcorrelations';
+const KEY_PAIR_SEPARATOR = '=';
+const PROPERTIES_SEPARATOR = ';';
+const ITEMS_SEPARATOR = ',';
+
+/* W3C Constrains*/
+
+export const MAX_NAME_VALUE_PAIRS = 180;
+export const MAX_PER_NAME_VALUE_PAIRS = 4096;
+export const MAX_TOTAL_LENGTH = 8192;
+
+/**
+ * Propagates {@link CorrelationContext} through Context format propagation.
+ *
+ * Based on the Correlation Context specification:
+ * https://w3c.github.io/correlation-context/
+ */
+export class HttpCorrelationContext implements HttpTextPropagator {
+  inject(context: Context, carrier: unknown, setter: SetterFunction) {
+    const distContext = getCorrelationContext(context);
+    if (distContext) {
```

Comment:
Instead of adding everything under `if` block, I would prefer to return immediately. Something like:
```
if (!correlationContext) return;
.. rest of the logic.
```

This is to be consistent with other propagator and increase the readability. WDYT?

Reply:
Agree, much better readability :)